api/extension-guides/chat.md

```diff
@@ -0,0 +1,204 @@
+
+# Summarize the whole topic in less than 300 characters for SEO purpose
+MetaDescription: A guide to creating an AI extension in Visual Studio Code
+---
+
+# Chat
+
+Visual Studio Code's Copilot Chat architecture allows extension authors to easily integrate with the Copilot Chat experience. And the unit of extensibility for Copilot Chat is an agent.
+
+Agents are domain experts that can answer the user query however they want - by fully using AI in the query processing or in a traditional way by forwarding it to a backend service. Agents can also provide the language model access to domain specific tools. With the help of the LLM, the agent may select a tool and define how to invoke it. VS Code ships with some built-in agents, for example `@workspace`. The `@workspace` agent knows about your workspace and can answer questions about it. Internally, the agent is powered by different tools: GitHub's knowledge graph combined with semantic search, local code indexes, and VS Code's language services.
+
+When a user explicitly mentions an `@agent` in their prompt, that prompt is forwarded to the extension that contributed that specific agent. Agents can respond using Markdown for simple text and image responses, or they can respond with a file tree or with buttons for a more interactive experience. For example, a file tree can be used as a preview when an agent is proposing to create a new workspace for the user. Agents can provide follow-ups for each response, imagine them as proposals on how to take the conversation further. To provide a smooth user experience, the whole API is streaming based. As already mentioned, agents can bring in sub commands - shortcuts to specific functionality. For example, the `@docker` agent might contribute a `/generate` sub command, resulting in the following example user prompt "`@docker /generate` a DOCKERFILE for workspace". The current syntax being explicit and concise can be a convenient time saver.
+
+![Chat concepts explanation](images/chat/chat.png)
+
+This screenshot shows the following chat concepts with a sample extension:
+1. `@cat` explicitly being called by the user via the `@` syntax
+2. Command explicitly called via the `/` syntax
+3. User query
+4. Copilot using the contributed `@cat` participant
+5. Markdown response that is fully controlled by the contributed `@cat` participant
+6. Code part of the markdown response
+7. [Button response](#response-stream)
+8. [Follow-ups](#follow-ups)
+9. Chat input for the user to continue. `description` of `@cat` is used as a placeholder
+
+
+## Chat Extension
+
+The [chat sample](https://github.com/microsoft/vscode-extension-samples/tree/main/chat-agent-sample) can be used as a starting point for your chat extension. It is a simple cat tutor that explains computer science topics using cat metaphors.
+
+![Diagram showing how extension can contribute to chat](images/chat/diagram.png)
+
+To start and register a chat participant you can do the following:
```

Comment:
This is an example of where I was missing where this piece of code fits into my extension. Let's say I'm new to building extensions and I want to extend Copilot Chat, what do I do after I've gone through the quickstart for creating an extension?

Reply:
I think you have tackled this one. Thus marking as resolved.